mode/oz/oz.js

```diff
@@ -0,0 +1,242 @@
+        return "number";
+
+      return null;
+    }
+
+    // Comments
+    if (ch == "%") {
+      stream.skipToEnd();
+      return 'comment';
+    }
+    else if (ch == "/") {
+      if (stream.eat("*")) {
+        state.tokenize = tokenComment;
+        return tokenComment(stream, state);
+      }
+    }
+
+    // Single operators
+    if(singleOperators.test(ch)) {
+      return "operator";
+    }
+
+    // If nothing match, we skip the entire alphanumerical block
+    stream.eatWhile(/\w/);
+
+    return "variable";
+  }
+
+  function tokenClass(stream, state) {
+    if (stream.eatSpace()) {
+      return null;
+    }
+    stream.match(/([A-Z][A-Za-z0-9_]*)|(`.+`)/);
+    state.tokenize = tokenBase;
+    return "variable-3"
+  }
+
+  function tokenMeth(stream, state) {
+    if (stream.eatSpace()) {
+      return null;
+    }
+    stream.match(/([a-zA-Z][A-Za-z0-9_]*)|(`.+`)/);
+    state.tokenize = tokenBase;
+    return "def"
+  }
+
+  function tokenFunProc(stream, state) {
+    if (stream.eatSpace()) {
+      return null;
+    }
+
+    if(!state.hasPassedFirstStage && stream.eat("{")) {
+      state.hasPassedFirstStage = true;
+      return "bracket";
+    }
+    else if(state.hasPassedFirstStage) {
+      stream.match(/([A-Z][A-Za-z0-9_]*)|(`.+`)|\$/);
+      state.hasPassedFirstStage = false;
+      state.tokenize = tokenBase;
+      return "def"
+    }
+    else {
+      state.tokenize = tokenBase;
+      return null;
+    }
+  }
+
+  function tokenComment(stream, state) {
+    var maybeEnd = false, ch;
+    while (ch = stream.next()) {
+      if (ch == "/" && maybeEnd) {
+        state.tokenize = tokenBase;
+        break;
+      }
+      maybeEnd = (ch == "*");
+    }
+    return "comment";
+  }
+
+  function tokenString(quote) {
+    return function (stream, state) {
+      var escaped = false, next, end = false;
+      while ((next = stream.next()) != null) {
+        if (next == quote && !escaped) {
+          end = true;
+          break;
+        }
+        escaped = !escaped && next == "\\";
+      }
+      if (end || !escaped)
+        state.tokenize = tokenBase;
+      return "string";
+    };
+  }
+
+  return {
+
+    startState: function () {
+      return {
+        tokenize: tokenBase,
+        currentIndent: 0,
+        doInCurrentLine: false,
+        hasPassedFirstStage: false
+      };
+    },
+
+    token: function (stream, state) {
+      if (stream.sol())
+        state.doInCurrentLine = 0;
+
+      return state.tokenize(stream, state);
+    },
+
+    indent: function (state, textAfter) {
+      var trueText = textAfter.replace(/^\s+|\s+$/g, '');
+
+      if (trueText.match(endKeywords) || trueText.match(middleKeywords) || trueText.match(/(\[])/))
+        return conf.indentUnit * (state.currentIndent - 1);
+
+      if (state.currentIndent < 0)
+        return 0;
+
+      return state.currentIndent * conf.indentUnit;
+    },
+    fold: "indent",
+    electricChars: "eEdDtTnNiIoOfFhHyY[]",
```

Comment:
Looks like you're trying to express something non-trivial here. You might want to use `electricInput` instead, which allows a regexp (which is matched starting from the start of the line) to reduce false positives.